### PR TITLE
Changed code to typeORM

### PR DESCRIPTION
**Pull request by OBaca**
- Changed code to TypeORM
- Added GET api/v1/trips
- Added GET api/v1/trips/{id}

### CODE REVIEW

File: server/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+    "cSpell.words": [
+        "Adumim",
+        "Afula",
```
Comment (OfekShimko) [inline server/.vscode/settings.json]: I think this vs code configuration file right? 
Do we all need that?

File: server/package.json [hunk 2]
```diff
@@ -10,18 +10,22 @@
...
     "typescript": "^5.6.3"
   },
   "dependencies": {
+    "axios": "^1.7.7",
     "bcryptjs": "^2.4.3",
     "dotenv": "^16.4.5",
     "express": "^4.21.1",
     "jsonwebtoken": "^9.0.2",
     "mysql2": "^3.11.4",
-    "tripnest-backend": "file:"
```
Comment (OfekShimko) [inline server/package.json:28]: What is this dependency?

File: server/src/entities/User.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+@Entity()
+export class User {
+  @PrimaryGeneratedColumn()
+  id!: number;
+
+  @Column()
+  username!: string;
+
+  @Column()
+  password!: string;
```
Comment (OfekShimko) [inline server/src/entities/User.ts:12]: WDYT?

```suggestion
  @Column()
  password!: string;

  // Lifecycle hook to hash the password before saving
  @BeforeInsert()
  @BeforeUpdate()
  async hashPassword(): Promise<void> {
    if (this.password) {
      const saltRounds = 10; // Number of salt rounds
      this.password = await bcrypt.hash(this.password, saltRounds);
    }
  }
```

File: server/src/entities/User.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';
+
+@Entity()
+export class User {
```
Comment (OfekShimko) [inline server/src/entities/User.ts:4]: Can we add email?

File: server/src/opentripmap.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+          lat,
+          lon,
+          radius: 5000,
+          types: type || 'interesting_places',
+          apikey: API_KEY,
+        }
+      });
+  
+      // Fetch detailed info for each activity
+      const activities = await Promise.all(
```
Comment (OfekShimko) [inline server/src/opentripmap.ts]: Nice

File: server/src/routes/userRoutes.ts [hunk 6]
```diff
@@ -1,56 +1,54 @@
...
-      } else {
-        res.status(500).json({ message: 'Error registering user' });
-      }
-    } catch (err) {
-      console.error(err);
-      res.status(500).json({ message: 'Database or server error' });
-    }
+
+    // Hash the password
+    const hashedPassword = await bcrypt.hash(password, 10);
```
Comment (OfekShimko) [inline server/src/routes/userRoutes.ts:53]: Ok I see you did it here, whatever you prefer

**Review by OfekShimko — APPROVED** [review]
Good job man!

File: server/.vscode/settings.json (hunk 1 above)
Comment (OBaca) [reply server/.vscode/settings.json]: we dont :)

File: server/src/entities/Trip.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';
+
+@Entity()
+export class Trip {
+  @PrimaryGeneratedColumn()
+  id!: number;
```
Comment (OfekShimko) [inline server/src/entities/Trip.ts]:
```suggestion
  @PrimaryGeneratedColumn('uuid')
  id!: string;
```

File: server/package.json (hunk 2 above)
Comment (OBaca) [reply server/package.json:28]: probably a mistake, I added more few changes

File: server/src/entities/User.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  @PrimaryGeneratedColumn()
+  id!: number;
```
Comment (OfekShimko) [inline server/src/entities/User.ts]:
```suggestion
  @PrimaryGeneratedColumn('uuid')
  id!: string;
```

File: server/src/entities/User.ts (hunk 4 above)
Comment (OBaca) [reply server/src/entities/User.ts:4]: Yes

File: server/src/routes/userRoutes.ts (hunk 6 above)
Comment (OBaca) [reply server/src/routes/userRoutes.ts:53]: You have other suggestions? Shay worked on it

**Comment by OBaca** [comment]
I added email to the users table

File: server/src/routes/userRoutes.ts (hunk 6 above)
Comment (OfekShimko) [reply server/src/routes/userRoutes.ts:53]: They are both good ( the comment I added on the DB entity is more strict). Like it is always make sure that the password is hashed ( on the DB level ). BUT for us it is fine, if you checked it with this code lets stay with it.

**Pull request merged**